pandas/tests/reshape/merge/test_merge.py

```diff
@@ -1369,12 +1364,115 @@ def f():
                             .09604978, -.06524096, .03532373,
                             .03025441, .036997, None, None]
             ))
-            .set_index(["household_id", "asset_id", "t"]))
+            .set_index(["household_id", "asset_id", "t"])
+            .reindex(columns=['share', 'log_return']))
 
-        def f():
-            household.join(log_return, how='outer')
+        result = (merge(household.reset_index(), log_return.reset_index(),
+                  on=['asset_id'], how='outer')
+                  .set_index(['household_id', 'asset_id', 't']))
+
+        assert_frame_equal(result, expected)
+
+
```

Comment:
I think you can move the multi test to a new file, e.g. reshape/merge/test_mult.py